runtime_scan/pkg/orchestrator/scanconfigwatcher/schedule.go

```diff
@@ -0,0 +1,264 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scanconfigwatcher
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aptible/supercronic/cronexpr"
+
+	"github.com/openclarity/vmclarity/api/models"
+)
+
+type ScheduleWindow struct {
+	start time.Time
+	end   time.Time
+}
+
+func (w *ScheduleWindow) Start() time.Time {
+	return w.start
+}
+
+func (w *ScheduleWindow) End() time.Time {
+	return w.end
+}
+
+func (w *ScheduleWindow) In(t time.Time) bool {
+	if t.Before(w.start) || t.After(w.end) {
+		return false
+	}
+
+	return true
+}
+
+func (w *ScheduleWindow) Before(t time.Time) bool {
+	return t.Before(w.start)
+}
+
+func (w *ScheduleWindow) After(t time.Time) bool {
+	return t.After(w.end)
+}
+
+func (w ScheduleWindow) Next() *ScheduleWindow {
+	return &ScheduleWindow{
+		start: w.end,
+		end:   w.end.Add(w.end.Sub(w.start)),
+	}
+}
+
+func (w ScheduleWindow) Prev() *ScheduleWindow {
+	return &ScheduleWindow{
+		start: w.start.Add(-1 * w.end.Sub(w.start)),
+		end:   w.start,
+	}
+}
+
+func (w ScheduleWindow) String() string {
+	return fmt.Sprintf("start: %s, end: %s", w.start.Format(time.RFC3339), w.end.Format(time.RFC3339))
+}
+
+// nolint:gomnd
+func NewScheduleWindow(now time.Time, size time.Duration) *ScheduleWindow {
+	return &ScheduleWindow{
```

Comment:
nit: it might be better to save the `size` to avoid calculating `w.end.Sub(w.start)` every  `Next()` and `Prev()`

Reply:
I'd keep it as is as it does not take a lot to compute Add/Sub on `int`.